satpy/readers/nc_nwcsaf.py

```diff
@@ -0,0 +1,160 @@
+from satpy.readers.file_handlers import BaseFileHandler
+
+logger = logging.getLogger(__name__)
+
+SENSOR = {'NOAA-19': 'avhrr/3',
+          'NOAA-18': 'avhrr/3',
+          'NOAA-15': 'avhrr/3',
+          'Metop-A': 'avhrr/3',
+          'Metop-B': 'avhrr/3',
+          'Metop-C': 'avhrr/3',
+          'EOS-Aqua': 'modis',
+          'EOS-Terra': 'modis',
+          'Suomi-NPP': 'viirs',
+          'JPSS-1': 'viirs', }
+
+PLATFORM_NAMES = {'MSG1': 'Meteosat-8',
+                  'MSG2': 'Meteosat-9',
+                  'MSG3': 'Meteosat-10',
+                  'MSG4': 'Meteosat-11', }
+
+
+class NcNWCSAF(BaseFileHandler):
+
+    """NWCSAF PPS&MSG NetCDF reader."""
+
+    def __init__(self, filename, filename_info, filetype_info):
+        """Init method"""
+        super(NcNWCSAF, self).__init__(filename, filename_info,
+                                       filetype_info)
+        self.nc = h5netcdf.File(filename, 'r')
+        self.pps = False
+
+        try:
+            # MSG:
+            sat_id = self.nc.attrs['satellite_identifier']
+            self.platform_name = PLATFORM_NAMES[sat_id]
+        except KeyError:
+            # PPS:
+            self.platform_name = self.nc.attrs['platform']
+            self.pps = True
+
+        self.sensor = SENSOR.get(self.platform_name, 'seviri')
+
+    # def get_shape(self, dsid, ds_info):
+    #     """Get the shape of the data."""
+    #     raise NotImplementedError
+    #     #     return self.nc[dsid.name].shape
+
+    def get_dataset(self, dsid, info, out=None):
+        """Load a dataset."""
+
+        logger.debug('Reading %s.', dsid.name)
+        variable = self.nc[dsid.name]
+
+        info = {'platform_name': self.platform_name,
+                'sensor': self.sensor}
+
+        try:
+            values = np.ma.masked_equal(variable[:],
+                                        variable.attrs['_FillValue'], copy=False)
+        except KeyError:
+            values = np.ma.array(variable[:], copy=False)
+        if 'scale_factor' in variable.attrs:
+            values = values * variable.attrs['scale_factor']
+            info['scale_factor'] = variable.attrs['scale_factor']
+        if 'add_offset' in variable.attrs:
+            values = values + variable.attrs['add_offset']
+            info['add_offset'] = variable.attrs['add_offset']
+
+        if 'valid_range' in variable.attrs:
+            info['valid_range'] = variable.attrs['valid_range']
+        if 'units' in variable.attrs:
+            info['units'] = variable.attrs['units']
+        if 'standard_name' in variable.attrs:
+            info['standard_name'] = variable.attrs['standard_name']
+
+        proj = Dataset(np.squeeze(values),
+                       copy=False,
+                       **info)
+        return proj
+
+    def get_area_def(self, dsid):
+        """Get the area definition of the datasets in the file. 
+        Only applicable for MSG products!"""
+        if self.pps:
+            # PPS:
```

Comment:
Is that enough to make sure it's not gridded data ? don't we have reprojected nwcsaf pps products ?